ledger-service/jwt/src/test/scala/com/digitalasset/jwt/JwksSpec.scala

```diff
@@ -69,6 +74,48 @@ class JwksSpec extends AnyFlatSpec with Matchers {
       s"The token with a mismatching public key should not successfully verify",
     )
   }
+
+  it should "invalidate cached key-id if verification failed" in new JwksSpec.Scope {
+    val token = generateToken("test-key-1", privateKey1)
+      .fold(e => fail("Failed to generate signed token: " + e.shows), x => x)
+    val result = verifier.verify(token)
+
+    assert(
+      result.isRight,
+      s"The correctly signed token should successfully verify, but the result was ${result.leftMap(e => e.shows)}",
+    )
+
+    server.stop(3)
+    //now `test-key-1` is cached
+
+    val updatedJwks = KeyUtils.generateJwks(
+      Map(
+        "test-key-2" -> publicKey2
+      )
+    )
+
+    val updatedServer = SimpleHttpServer.start(updatedJwks, Some(server.getAddress.getPort))
+
+    val expiredToken = generateToken("test-key-1", privateKey1, secondsTillExpiration = -100)
```

Comment:
Wouldn't it be better if this failure was done with `test-key-2` instead, to drive the idea home that any misvalidation triggers cache purge?